src/main/java/bio/terra/service/common/JournalCreateDeleteEntryStep.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.service.common;
+
+import static bio.terra.service.common.CommonFlightUtils.getFlightInformationOfInterest;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.auth.iam.IamResourceType;
+import bio.terra.service.journal.JournalService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class JournalCreateDeleteEntryStep implements Step {
+  private final JournalService journalService;
+  private final AuthenticatedUserRequest userReq;
+  private final UUID resourceKey;
+  private final IamResourceType resourceType;
+  private final String note;
+
+  public JournalCreateDeleteEntryStep(
+      JournalService journalService,
+      AuthenticatedUserRequest userRequest,
+      UUID resourceKey,
+      IamResourceType resourceType,
+      String note) {
+    this.journalService = journalService;
+    this.userReq = userRequest;
+    this.resourceKey = resourceKey;
+    this.resourceType = resourceType;
+    this.note = note;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    journalService.journalDelete(
```

Comment:
I think we should have something that removes the entry if the step is undone...then you're going to end end up with a bunch of false entries.  Or is the idea to join with the flight so that users can see that the entry was associated with something that went wrong?

Reply:
By convention, I've used these Stairway "Journaling" helpers as the last steps in Stairway flights so they don't have to be unwound if any of the prior steps fail.

I could introduce a return value for each journal action (journalCreate, journalEdit, journalDelete) and add a method for callers to remove journal entries.  Generally speaking, I was trying to avoid this - because a user might see an entry disappear.  Would you like me to move forward with that approach?

Reply:
I'm not too concerned with the jitter (we could even add a thing to make sure to only show entries where associated flights are ended).  I guess my main concern is if we eventually have a need to have, say 2 journal calls in a flight...I think it would make sense to roll them both back.